src/mlfe_ast.hrl

```diff
@@ -77,7 +77,7 @@
              | t_map()
              | t_tuple()
              | t_clause()
-             | typer:t_cell().  % a reference cell for a type.
+             | mlfe_typer:t_cell().  % a reference cell for a type.
```

Comment:
This is the line introducing the dialyzer errors. You get the same thing on master if you change the same line. Will try to fix the warnings in a separate branch.